### PR TITLE
Custom asset names

### PR DESCRIPTION
**Pull request by EmileSonneveld**
Implementing "filename_prefix" to prefix outputted asset names.

### CODE REVIEW

File: tests/test_download.py [hunk 1]
```diff
@@ -195,78 +224,268 @@ def test_download_masked_json_reproject(self, tmp_path):
...
+                # "multidate": True,  # not used
+            }
+        )
+        # with open(self.test_write_assets_parameterize_path + test_name + ".json", 'w') as fp:
+        #     json.dump(assets, fp, indent=2)
+
+        name, asset = next(iter(assets.items()))
+        print("href of first asset: " + asset['href'])
+
+        # Special case for saveRDDTemporal as it only returns one case, but is designed to return multiple cases
```
Comment (EmileSonneveld) [inline tests/test_download.py:464]: @jdries Here I was not sure why `saveRDDTemporal` only returns one file, while other code paths returned multiple files.
This is not something I changed, but only added coverage for. So, ok to just document it like this?

Comment (jdries) [reply tests/test_download.py:464]: from what I can tell on sight, create_spacetime_layer creates a layer with only one date (self.now). Hence I would indeed inspect only one asset.
Test coverage would indeed be better if we have more than one date, can we change that method?

Comment (EmileSonneveld) [reply tests/test_download.py]: That was indeed it. The hacky if-statement is removed now.

File: tests/test_download.py [hunk 2]
```diff
@@ -89,15 +103,41 @@ def create_spacetime_layer(self):
 
         return TiledRasterLayer.from_numpy_rdd(LayerType.SPACETIME, rdd, metadata)
 
+    def create_spatial_layer(self):
```
Comment (jdries) [inline tests/test_download.py:106]: could this be moved to datacube_fixtures.py, or even reuse an existing function from somewhere?

**Review by jdries — CHANGES_REQUESTED** [review]
reviewed, only one minor comment

File: tests/test_download.py (hunk 2 above)
Comment (EmileSonneveld) [reply tests/test_download.py]: Done. If Ok, I can merge

**Pull request merged**